schema/revert/mutations/create_project_revision.sql

```diff
@@ -1,18 +1,25 @@
--- Revert cif:mutations/create_project_revision from pg
+-- Deploy cif:mutations/create_project_revision to pg
 
 begin;
 
-drop function cif.create_project_revision(integer, varchar(1000), varchar(1000)[]);
+drop function cif.create_project_revision(project_id integer);
 
-create or replace function cif.create_project_revision(project_id integer)
+create or replace function cif.create_project_revision(project_id integer, revision_type varchar(1000) default 'Amendment', amendment_types varchar(1000)[] default array[]::varchar[])
 returns cif.project_revision
 as $function$
 declare
   revision_row cif.project_revision;
   form_change_record record;
+  _amendment_type varchar(1000);
 begin
-  insert into cif.project_revision(project_id, change_status)
-  values ($1, 'pending') returning * into revision_row;
+  insert into cif.project_revision(project_id, change_status, revision_type)
+  values ($1, 'pending', $2) returning * into revision_row;
+
+  foreach _amendment_type in array $3
+    loop
+      insert into cif.project_revision_amendment_type(project_revision_id, amendment_type)
+      values (revision_row.id, (select name from cif.amendment_type where cif.amendment_type.name = _amendment_type));
+    end loop;
```

Comment:
This could be refactored to a single query (for loops are frowned upon in SQL, the query engine is much better and efficient at retrieving and building sets). 

Something like this:

```suggestion
      insert into cif.project_revision_amendment_type(project_revision_id, amendment_type)
      values (
          select 
              revision_row.id as project_revision_id, 
              name as amendment_type 
           from cif.amendment_type 
           where cif.amendment_type.name in $3
       );
```

We could put this as a tech debt card to refactor later if it's too time consuming, since we only have 3 amendment types for now